vmware-event-router/internal/processor/aws_event_bridge.go

```diff
@@ -53,13 +55,18 @@ type eventPattern struct {
 
 // NewAWSEventBridgeProcessor returns an AWS EventBridge processor for the given
 // stream source.
-func NewAWSEventBridgeProcessor(ctx context.Context, cfg connection.Config, source string, verbose bool, ms *metrics.Server) (Processor, error) {
+func NewAWSEventBridgeProcessor(ctx context.Context, cfg connection.Config, ms metrics.Receiver, opts ...AWSOption) (Processor, error) {
```

Comment:
This is one massive construction function. Not only does it have multiple switch statements and for loops, it also runs two goroutines. I would recommend breaking this function up, maybe putting the section finding the ARN rules in another function. But, I can see the function works, so I'm also fine with keeping it. Even if it's a long function, it's not too difficult to follow what's going on.

Reply:
Fair point, even though I've seen larger constructors in the wild :) I definitely think breaking out the options handling and pattern match logic into helpers makes sense, especially since the `syncRules` has a lot of repetitive code here.

Reply:
Agreed that the change would better suited to be in a separate PR. I have raised a new issue to have the matter addressed: https://github.com/vmware-samples/vcenter-event-broker-appliance/issues/168